docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,374 @@
+)
+
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {
+    // Create the proposal handler that will be used to fill proposals with
+	// transactions and oracle data.
+	proposalHandler := proposals.NewProposalHandler(
+		app.Logger(),
+		baseapp.NoOpPrepareProposal(),
+		baseapp.NoOpProcessProposal(),
+		ve.NewDefaultValidateVoteExtensionsFn(app.StakingKeeper),
+		compression.NewCompressionVoteExtensionCodec(
+			compression.NewDefaultVoteExtensionCodec(),
+			compression.NewZLibCompressor(),
+		),
+		compression.NewCompressionExtendedCommitCodec(
+			compression.NewDefaultExtendedCommitCodec(),
+			compression.NewZStdCompressor(),
+		),
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+		oracleMetrics,
+	)
+	app.SetPrepareProposal(proposalHandler.PrepareProposalHandler())
+	app.SetProcessProposal(proposalHandler.ProcessProposalHandler())
+}
+```
+
+Next, set up the `PreBlocker`. This involves:
+
+- **Aggregate Function:** Setting the aggregator function that combines all reported prices into one final price per currency pair.
+- **Currency Pair Strategy:** Setting the currency pair strategy. For this example, we will use the `DeltaCurrencyPairStrategy` which encodes/decodes the price as the difference between the current price and the previous price.
+- **Data Compression Codecs:** Setting the compression strategy for vote extensions and extended commits.
+
+```go oracle.go
+package app
+
+import (
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+
+    oraclepreblock "github.com/skip-mev/slinky/abci/preblock/oracle"
+    compression "github.com/skip-mev/slinky/abci/strategies/codec"
+    "github.com/skip-mev/slinky/abci/strategies/currencypair"
+    "github.com/skip-mev/slinky/pkg/math/voteweighted"
+)
+
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {
+	// ... snip ...
+
+	// Create the aggregation function that will be used to aggregate oracle data
+	// from each validator.
+	aggregatorFn := voteweighted.MedianFromContext(
+		app.Logger(),
+		app.StakingKeeper,
+		voteweighted.DefaultPowerThreshold,
+	)
+	veCodec := compression.NewCompressionVoteExtensionCodec(
+		compression.NewDefaultVoteExtensionCodec(),
+		compression.NewZLibCompressor(),
+	)
+	ecCodec := compression.NewCompressionExtendedCommitCodec(
+		compression.NewDefaultExtendedCommitCodec(),
+		compression.NewZStdCompressor(),
+	)
+
+	// Create the pre-finalize block hook that will be used to apply oracle data
+	// to the state before any transactions are executed (in finalize block).
+	oraclePreBlockHandler := oraclepreblock.NewOraclePreBlockHandler(
+		app.Logger(),
+		aggregatorFn,
+		app.OracleKeeper,
+		oracleMetrics,
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+		veCodec,
+		ecCodec,
+	)
```

Comment:
Looking at all of this, I think we should eventually just make something like 

```go 
func DefaultConstructor(app, stakingKeeper, oracleKeeper, metrics) preblocker 
```

Reply:
we should really be abstracting away most of this so our integrations can be less LoC.  We can document this as much as we want, but the current construction of everything is probably more complicated than it needs to be

Reply:
opened https://linear.app/skip/issue/BLO-1576/create-default-constructor-for-connect-integrations